### PR TITLE
add filtering and paging

### LINKED ISSUE #34: Pagination and Filtering for recipes
Filter recipes by:
- Multiple Tags
- Name (use the LIKE operator)
- Author Id

Apply pagination (skip top)

### PR DESCRIPTION
**Pull request by Rytis-Sapka**
added filtering and paging
also reintroduced Recipes repository to DI container, since it was now using the old generic repository.

Closes #34

### CODE REVIEW

**Comment by 6ar8nas** [comment]
Looks good to me, though I'd wait for some other eyes to look at it.

File: Recipes/Recipes.API/Controllers/RecipesController.cs [hunk 1]
```diff
@@ -27,18 +28,35 @@ public class RecipesController : ControllerBase
...
     [ProducesResponseType(StatusCodes.Status500InternalServerError)]
-    public async Task<ActionResult<IEnumerable<Recipe>>> GetAllRecipes()
+    public ActionResult<IEnumerable<Recipe>> GetAllRecipes([FromQuery] string? title, [FromQuery] string? csvTags, [FromQuery] long? authorId, [FromQuery] int? skip, [FromQuery] int? top)
     {
-        IEnumerable<RecipeEntity> recipes;
+        IQueryable<RecipeEntity> recipesQuery;
         try
         {
-            recipes = await _recipesRepository.GetAllAsync();
+            recipesQuery = _recipesRepository.GetQueryable();
```
Comment (BenasB) [inline Recipes/Recipes.API/Controllers/RecipesController.cs:36]: You just made the repository pattern useless :D

**Comment by adomas-vensas** [comment]
Since there is no other comments and this feaeture works, I will approve

**Review by adomas-vensas — APPROVED** [review]

**Review by BenasB — APPROVED** [review]
I assume you have tested this, so I'll turn a blind eye as long as we can get the functionality on front end

File: Recipes/Recipes.API/Controllers/RecipesController.cs (hunk 1 above)
Comment (Rytis-Sapka) [reply Recipes/Recipes.API/Controllers/RecipesController.cs]: I guess I got a little caried away putting everything in controllers. 
Moved filtering logic to Repository

**Pull request merged**